core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1323,17 +1321,17 @@ class IsrChangeNotificationListener(controller: KafkaController) extends IZkChil
 
       // delete processed children
       childrenAsScala.map(x => ZkUtils.deletePath(controller.controllerContext.zkClient,
-                                                  ZkUtils.getEntityConfigPath(ConfigType.Topic, x)))
+                                                  ZkUtils.IsrChangeNotificationPath + "/" + x))
```

Comment:
Perhaps it's better to do this in a finally clause so that in case we hit any unexpected json parsing error, the isr change path is still deleted.